src/ct_eval.cpp

```diff
@@ -11,89 +11,98 @@ namespace TypeChecker {
 
 // literals
 
-Own<TypedAST::ArrayLiteral> ct_eval(Own<TypedAST::ArrayLiteral> ast, TypeChecker& tc) {
-	for (auto&& element : ast->m_elements)
-		element = ct_eval(std::move(element), tc);
+TypedAST::ArrayLiteral* ct_eval(
+    TypedAST::ArrayLiteral* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
+	for (auto& element : ast->m_elements)
+		element = ct_eval(element, tc, alloc);
 	return ast;
 }
 
 // expressions
 
-Own<TypedAST::FunctionLiteral> ct_eval(
-    Own<TypedAST::FunctionLiteral> ast, TypeChecker& tc) {
+TypedAST::FunctionLiteral* ct_eval(
+    TypedAST::FunctionLiteral* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
 
 	// TODO: type hints
 
 	assert(ast->m_body->type() == TypedASTTag::Block);
-	auto body = static_cast<TypedAST::Block*>(ast->m_body.get());
-	for (auto&& child : body->m_body)
-		child = ct_eval(std::move(child), tc);
+	auto body = static_cast<TypedAST::Block*>(ast->m_body);
+	for (auto child : body->m_body)
```

Comment:
iterate by ref

Reply:
This one I completely missed it. Probably a misclick in vim.